lib/reversetunnel/agent.go

```diff
@@ -585,6 +588,19 @@ func (a *agent) handleChannels() error {
 				a.handleDiscovery(ch, req)
 				a.wg.Done()
 			}()
+		case nch := <-a.reconnectC:
+			if nch == nil {
+				continue
+			}
+
+			a.log.Debugf("Received reconnect from proxy.")
+			c, _, err := nch.Accept()
+			if err != nil {
+				return trace.Wrap(err)
+			}
+
+			c.Close()
+			return nil
```

Comment:
Can we call `a.Stop()` here instead of exiting this loop (maybe after setting `a.reconnectC` to `nil` or something to avoid stopping more than once)? We want the agentpool to start disregarding this connection but we still want it to be serviced correctly, in case some connections are about to be opened or something like that.

Reply:
Move to a global request and call stop. Stop changes the state to closed but leaves the connection open while open transports drain so we should be all set.